bot/src/embeds/pnw.rs

```diff
@@ -21,15 +21,12 @@ pub fn alliance<'a>(
         .data()
         .cache
         .find_many_nations(|a| a.alliance_id == alliance.id);
-    let cities = ctx.data().cache.find_many_cities(|a| {
-        nations.contains(
-            &ctx.data()
-                .cache
-                .find_exactly_one_nation(|n| n.id == a.nation_id)
-                .unwrap(),
-        )
-    });
 
+    let nation_ids = nations.iter().map(|a| a.id).collect::<Vec<i32>>();
```

Comment:
Maybe use a HashSet here, O(1) contains instead of O(n)

Reply:
done